### DISCUSSION TITLE
Consider Modular's Mojo language support

### DISCUSSION THREAD

**Discussion by chopper2000uk** [godotengine/godot-proposals#7726] [Scripting]
It might not be feasible or bring any benefits, and it's very early days, Mojo is only in a very early release and not even open sourced as yet. 
But it might be worth reviewing as to whether it could be an option as a first class supported language within the IDE and what benefits it may or may not bring, and if it could even be a possible future successor to GDScript.

https://docs.modular.com/mojo/why-mojo.html

**Comment by LaurentOngaro** [comment]
I've just discovered MOJO and it's very promising, and it seems to share some features of GDscript.

Some key features:
- superset of python and 100% compatible with existing code (but not for now)
- code speed can be significantly improved with compilation, AOT or JIT
- static typing can also be used as an option to improve code speed
- use "fn" instead of "def" to declare "statically typed" functions and gain speed and "type and code checking" at compile time
- multithreading and parallelization are essential features
- a project can be compiled as an exe (not possible in python without major drawbacks)
- still in beta for now
- proprietary code at startup (to speed up the startup process) but open source whenever possible

This could really be an alternative to have the speed of C++ while remaining close to GDscript syntax. No more need to use two code syntaxes if speed is important

**Reply by AThousandShips** [reply]
> superset of python and 100% compatible with existing code 

GDScript is not a subset of Python so there's no guarantee of this, GDScript has it's own syntax, it only resembles Python

**Comment by AThousandShips** [comment]
There are no plans to replace GDScript and likely won't be, and this can't really be even considered as an alternative *before* it becomes open source, the fact that it's not open source *yet* is not a good indication to me at least for it being guaranteed to be so (in fact the website makes no statements on it being planned to be open source)

**Reply by Zireael07** [reply]
> in fact the website makes no statements on it being planned to be open source)

You need to click around a bit but it DOES make such statements

https://docs.modular.com/mojo/faq.html but you need to scroll pretty far down

**Comment by frankcarey** [comment]
As an update: a website post from March, 28th 2024 says it's now open source. (The core modules and standard lib parts anyways). 

> The Next Big Step in Mojo🔥 Open Source

> At Modular, open source is ingrained in our DNA. We firmly believe for Mojo to reach its full potential, it must be open source. We have been progressively open-sourcing more of Mojo and parts of the MAX platform, and today we’re thrilled to announce the release of the core modules from the Mojo standard library under the Apache 2 license!

> We have always believed that building Mojo in the open will lead to a better result, because it allows its design to be shaped by the feedback from the broader community. We released Mojo very early and have been driving steady improvements since May of 2023 (see the changelog). Building a language and its infrastructure is hard work and takes time, and we’re excited to move from sharing our work to collaborating with Mojo developers worldwide.

https://mojolang.org/
